internal/otc_wrapper.go

```diff
@@ -34,16 +34,21 @@ func NewOtcClientFromConfig(config ConfigStruct) (*OtcWrapper, error) {
 	return &OtcWrapper{providerClient: provider, Region: string(config.AuthenticationData.Region)}, nil
 }
 
-func (c *OtcWrapper) GetMetrics() ([]otcMetrics.MetricInfoList, error) {
+func (c *OtcWrapper) GetMetrics(logger ILogger) ([]otcMetrics.MetricInfoList, error) {
 	opts := golangsdk.EndpointOpts{Region: c.Region}
 	cesClient, err := openstack.NewCESClient(c.providerClient, opts)
 	if err != nil {
 		return []otcMetrics.MetricInfoList{}, err
 	}
-	metricsResponsePages, err := otcMetrics.ListMetrics(cesClient, otcMetrics.ListMetricsRequest{}).AllPages()
+
+	// Explicitly set limit to the maximum allowed, as the default seems to not be set (should be 1000, currently is nothing)
+	limit := 1000
```

Comment:
The situation here is that the request is taking a pointer to an int. This is an easy way to make this value optional. The struct has a struct annotation with "ommitempty". This means that the rendered JSON will not contain this field. I expect the service to be able to understand that if this value is not set to set the default value.

Does this make sense to you?

Reply:
Yeah, this is what I thought as well, but when I checked I couldn't find where the default was being set in the SDK. Without explicitly setting the limit here like this, the pager only returns the first page, as limit is empty. At least, that's how it was working on my local machine.

I was missing metrics without setting the limit, but with setting the limit I get all the metrics we have.

Reply:
Alright, then let's keep it. I will also run it today once and check the debugger.